mars/oscar/backends/ray/pool.py

```diff
@@ -369,11 +369,3 @@ async def check_main_pool_alive(self, main_pool):
                 "Main pool %s has exited, exit current sub pool now.", main_pool
             )
             os._exit(0)
-
-
-if ray and ray.is_initialized():
-    # When using ray client to connect to a ray cluster, ray server will act as mars driver. All mars call from mars
-    # client will go to ray server first, then the ray server will ray call to other actors. So the ray server need to
-    # register ray serializers.
-    # TODO Need a way to check whether current process is a ray server.
-    register_ray_serializers()
```

Comment:
If ray client server doesn't `register_ray_serializers`, mars objects serialization may get error?

Reply:
Especially for ray ObjectRef and ActorHandle, since thay are serialized by pickler

Reply:
> If ray client server doesn't `register_ray_serializers`, mars objects serialization may get error?

Currently, there is no case covers this fix. I believe these lines fix some issues, but current implementation is too rough.

I think we can find a better solution after this PR. For example, 
- `register_ray_serializers()` only when necessary.
- Add some test cases cover this fix.

Reply:
If ray client server is started at a different process with driver, skip `register_ray_serializers` in client will cause serializatin issue. Curretn test case `test_ray.py::test_ray_client` didn't start a ray client server in another process, this is why removing it didn't throws error.

Reply:
I think we can skip `register_ray_serializers` issue in this PR and fix it in next PR.

Reply:
Thanks. I will create a new PR with fixes and test cases.